delegation-mock/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+#![no_std]
+
+multiversx_sc::imports!();
+
+pub type Epoch = u64;
+pub const MAX_PERCENTAGE: u64 = 100_000;
+pub const APY: u64 = 10_000; //10%
+pub const EPOCHS_IN_YEAR: u64 = 365;
+pub const UNBOND_PERIOD: u64 = 10;
+
+pub mod delegation_proxy;
+
+#[multiversx_sc::contract]
+pub trait DelegationMock {
+    #[init]
+    fn init(&self) {}
+
+    #[payable("EGLD")]
+    #[only_owner]
+    #[endpoint(depositEGLD)]
+    fn deposit_egld(&self) {
+        let payment_amount = self.call_value().egld_value().clone_value();
+        self.egld_token_supply()
+            .update(|value| *value += &payment_amount);
+    }
+
+    #[payable("EGLD")]
+    #[endpoint(delegate)]
+    fn delegate(&self) {
+        let caller = self.blockchain().get_caller();
+        let payment_amount = self.call_value().egld_value().clone_value();
+        self.address_deposit(&caller)
+            .update(|value| *value += &payment_amount);
+        self.egld_token_supply()
+            .update(|value| *value += &payment_amount);
+    }
+
+    #[endpoint(unDelegate)]
+    fn undelegate(&self, egld_to_undelegate: BigUint) {
+        let caller = self.blockchain().get_caller();
+        let current_epoch = self.blockchain().get_block_epoch();
+        let total_deposit = self.address_deposit(&caller).get();
+        require!(
+            egld_to_undelegate > 0 && egld_to_undelegate <= total_deposit,
+            "Invalid undelegate amount"
+        );
+        self.address_deposit(&caller)
+            .update(|value| *value -= &egld_to_undelegate);
+        self.address_undelegate_amount(&caller)
+            .update(|value| *value += &egld_to_undelegate);
+        self.address_undelegate_epoch(&caller)
+            .set(current_epoch + UNBOND_PERIOD);
+    }
+
+    #[endpoint(withdraw)]
```

Comment:
If the endpoint name is the same as the function name, you can simply use `#[endpoint]` annotation.